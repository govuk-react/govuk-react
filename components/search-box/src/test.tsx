import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import { SearchBox } from '.';

const example = 'example';
const wrapper = (
  <SearchBox>
    <SearchBox.Input placeholder={example} />
    <SearchBox.Button />
  </SearchBox>
);

describe('SearchBox', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
