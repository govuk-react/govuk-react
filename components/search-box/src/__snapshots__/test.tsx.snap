// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox matches wrapper snapshot: wrapper mount 1`] = `
.c1 {
  width: 100%;
  height: 40px;
  padding: 6px;
  margin: 0;
  border: solid 1px #bfc1c3;
  border-right: 0;
  box-sizing: border-box;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.75;
  background: #ffffff;
  border-radius: 0;
  -webkit-appearance: none;
}

.c1:focus {
  margin-right: 3px;
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c1:focus ~ button {
  width: 46px;
}

.c2 {
  background-color: #2b8cc4;
  border: 0;
  display: block;
  color: #ffffff;
  position: relative;
  padding: 10px;
  width: 45px;
  height: 40px;
  background-repeat: no-repeat;
  background-position: 2px 50%;
  text-indent: -999em;
  overflow: hidden;
}

.c2:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  background: #ffffff;
  margin-bottom: 0;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<SearchBox>
  <src__StyledSearchBox>
    <div
      className="c0"
    >
      <SearchBox.Input
        placeholder="example"
        type="search"
      >
        <input
          className="c1"
          placeholder="example"
          type="search"
        />
      </SearchBox.Input>
      <SearchBox.Button>
        <src__SearchButton
          title="Search"
        >
          <button
            className="c2"
            title="Search"
          >
            <Search
              fill="#ffffff"
              title="Search"
            >
              <SVG
                fill="#ffffff"
                title="Search"
                viewBox="0 0 57 57"
              >
                <svg
                  fill="#ffffff"
                  height="100%"
                  style={
                    Object {
                      "display": "block",
                    }
                  }
                  version="1.1"
                  viewBox="0 0 57 57"
                  width="100%"
                >
                  <title>
                    Search
                  </title>
                  <path
                    d="M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23 s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92 c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17 s-17-7.626-17-17S14.61,6,23.984,6z"
                  />
                </svg>
              </SVG>
            </Search>
          </button>
        </src__SearchButton>
      </SearchBox.Button>
    </div>
  </src__StyledSearchBox>
</SearchBox>
`;
